pallets/xor-fee/src/lib.rs

```diff
@@ -830,7 +765,43 @@ where
 }
 
 impl<T: Config> Pallet<T> {
-    pub fn remint_val(xor_to_val: Balance) -> Result<(), DispatchError> {
+    pub fn random_remint() -> Weight {
+        let period = RemintPeriod::<T>::get();
+        let mut weight = Weight::default();
```

Comment:
Looks like need add 1 read for getting period
and in Ok match 2 reads and writes for take